src/video/uikit/SDL_uikitviewcontroller.m

```diff
@@ -684,20 +668,20 @@ bool UIKit_HasScreenKeyboardSupport(SDL_VideoDevice *_this)
     return true;
 }
 
-void UIKit_ShowScreenKeyboard(SDL_VideoDevice *_this, SDL_Window *window, SDL_PropertiesID props)
+bool UIKit_StartTextInput(SDL_VideoDevice *_this, SDL_Window *window, SDL_PropertiesID props)
 {
     @autoreleasepool {
         SDL_uikitviewcontroller *vc = GetWindowViewController(window);
-        [vc setKeyboardProperties:props];
-        [vc showKeyboard];
+        [vc setTextFieldProperties:props];
```

Comment:
It seems doing this here is not a good idea, if text input is active but `StartTextInput` is called again with different properties then the new properties are not applied because text input is already active. I guess this explains why this existed in `ShowScreenKeyboard` (and why that function takes text input properties).

Thoughts on how this should be fixed? I'm thinking about defining a new function in `SDL_VideoDevice` that is solely responsible for updating the text input properties (call it `SetTextInputProperties`), and have that function be called on `StartTextInput` regardless of text input active state or keyboard visibility state.

Reply:
That seems reasonable, and can be done as a separate commit.

Reply:
> It may be that on some (all?) platforms the on-screen keyboard needs to be hidden and shown with the new properties when they change.

Well, since `SDL_StartTextInput` is the one accepting text input properties, I think it's sort of expected for SDL to handle the case where text input is already active. I think the approach of a separate method for updating text properties is flexible enough to handle both platforms that can update without hiding the keyboard (by directly updating the required states), and platforms that require hiding the keyboard first (by internally hiding and showing the keyboard with the new properties).

Reply:
Actually looking at the X11/steamdeck implementation, I think I'm better off moving `setTextFieldProperties` back to be called on showing on-screen keyboard and call it a day.

Reply:
Never mind I misread again, `ShowScreenKeyboard` does not get called if the keyboard is shown already...

I guess I'll add the method and implement it only on iOS for now.